File: 4284-smallest-stable-index-i/smallest-stable-index-i.java
class Solution {
    public int firstStableIndex(int[] nums, int k) {
        int mx=Integer.MIN_VALUE;
        int mn=Integer.MAX_VALUE;
        int len=nums.length;
        int help[] = new int[len];
        for(int i=len-1;i>=0;i--){
            mn=Math.min(mn,nums[i]);
            help[i]=mn;
        }
        for(int i=0;i<len;i++){
            mx=Math.max(mx,nums[i]);
            int maxE=mx;
            int minE=help[i];
            if(maxE-minE<=k){
                return i;
            }
        }
        return -1;
    }
}

File: 4284-smallest-stable-index-i/smallest-stable-index-i.py
class Solution:
    def firstStableIndex(self, nums: list[int], k: int) -> int:
        mx=0;mn=0
        for i in range(len(nums)):
            mx=max(nums[:i+1])
            mn=min(nums[i:])
            cv=mx-mn
            if cv<=k:
                return i
        return -1